aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/UploadOptions.java

```diff
@@ -0,0 +1,124 @@
+package com.amazonaws.mobileconnectors.s3.transferutility;
+
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+
+/**
+ * A class that can hold all of the optional parameters (bucket, metadata, cannedAcl, transferListener).
+ */
+public class UploadOptions {
```

Comment:
Make sure to add `equals`, `hashCode`, and `toString` to this class, in case customers ever do any comparisons with this object.  

These are pretty easy to add with Android Studio - you can type Command + N, and select `equals & hashCode`, and then `toString`.

Reply:
Ooh I didn't know about the ⌘-N!

Reply:
If you are using an apple keyboard, it should be 'Control + N' 😄